cardano-smash-server/src/Cardano/SMASH/Server/Run.hs

```diff
@@ -16,27 +16,35 @@ import           Network.Wai.Handler.Warp (defaultSettings, runSettings, setBefo
 
 import           Cardano.BM.Trace (Trace, logInfo)
 
-import           Cardano.Db (PGPassSource (..), textShow)
+import           Cardano.Db (PGPassSource (PGPassDefaultEnv), readPGPass, renderPGPassError,
+                   textShow, toConnectionString)
+import qualified Cardano.Db as Db
 
 import           Cardano.SMASH.Server.Api
 import           Cardano.SMASH.Server.Config
 import           Cardano.SMASH.Server.Impl
 import           Cardano.SMASH.Server.PoolDataLayer
 import           Cardano.SMASH.Server.Types
 
+import           Control.Monad.Trans.Except.Exit (orDie)
+import           Control.Monad.Trans.Except.Extra (newExceptT)
+import           Database.Persist.Postgresql (withPostgresqlPool)
+
 runSmashServer :: SmashServerConfig -> IO ()
 runSmashServer config = do
 
     let trce = sscTrace config
-    let poolDataLayer = postgresqlPoolDataLayer trce PGPassDefaultEnv
-
     let settings =
           setPort (sscSmashPort config) $
           setBeforeMainLoop
             (logInfo trce $ "SMASH listening on port " <> textShow (sscSmashPort config))
           defaultSettings
 
-    runSettings settings =<< mkApp (sscTrace config) poolDataLayer (sscAdmins config)
+    pgconfig <- orDie renderPGPassError $ newExceptT (readPGPass PGPassDefaultEnv)
+    Db.runIohkLogging trce $ withPostgresqlPool (toConnectionString pgconfig) defaultSmashPool $ \pool -> do
```

Comment:
Shouldn't this use `sspPsqlPool config` instead of `defaultSmashPool` to make it actually configurable.